plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/conditions/PropertyConditionEvaluator.java

```diff
@@ -206,7 +206,7 @@ protected boolean isMatch(String op, Object actualValue, String expectedValue, O
         if (op == null) {
             return false;
         } else if (actualValue == null) {
-            return op.equals("missing");
+            return op.equals("missing")|| op.equals("notIn") || op.equals("notEquals");
```

Comment:
What happen if:

expectedValue: null
actualValue: null
op: notEquals

It's look like it will return true, but it should return false in that case.

Reply:
Thanks @jkevan 
i dont think we can have both  expectedValue=null && op=notEquals
like that:
{
        "type": "profilePropertyCondition",
        "parameterValues": {
          "propertyName": "properties.countryCode",
          "comparisonOperator": "notEquals",
          "propertyValue": "null"
        }
}

in such case we would use comparisonOperator=missing
{
        "type": "profilePropertyCondition",
        "parameterValues": {
          "propertyName": "properties.countryCode",
          "comparisonOperator": "missing"
        }
}

Reply:
sorry for the delay, from what i tested, then yes in elastic it handles it this way.
the way i test it is have 2 tests:
1. create a segment condition,  then send an event that matches, after validates that the profile is in the segment.
2. create the event first and then create the segment

Reply:
Thx for the clarification

Reply:
But I also found this case that should be handle the same: 'hasNoneOf'

```suggestion
            return op.equals("missing")|| op.equals("notIn") || op.equals("notEquals") || op.equals("hasNoneOf");
```

Reply:
Hi, sorry im working on a different project at the moment and i will not have time to push new commits